parity/sync/main.rs

```diff
@@ -0,0 +1,135 @@
+// Copyright 2015, 2016 Ethcore (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Parity sync service
+
+extern crate ethcore_ipc_nano as nanoipc;
+extern crate ethcore_ipc_hypervisor as hypervisor;
+extern crate ethcore_ipc as ipc;
+extern crate ctrlc;
+#[macro_use] extern crate log;
+extern crate ethsync;
+extern crate rustc_serialize;
+extern crate docopt;
+extern crate ethcore;
+extern crate ethcore_util as util;
+
+use std::sync::Arc;
+use hypervisor::{HypervisorServiceClient, SYNC_MODULE_ID, HYPERVISOR_IPC_URL};
+use ctrlc::CtrlC;
+use std::sync::atomic::{AtomicBool, Ordering};
+use docopt::Docopt;
+use ethcore::client::{RemoteClient, ChainNotify};
+use ethsync::{SyncProvider, SyncConfig, EthSync, ManageNetwork, NetworkConfiguration};
+use std::thread;
+use util::numbers::{U256, H256};
+use std::str::FromStr;
+use nanoipc::IpcInterface;
+
+const USAGE: &'static str = "
+Ethcore sync service
+Usage:
+  sync <client-url> <network-id> <listen-address> <nat-enabled> <discovery-enabled> <ideal-peers> <config-path> <allow-non-reserved> [options]
+
+Options:
+  --public-address IP      Public address.
+  --boot-nodes LIST        List of boot nodes.
+  --reserved-nodes LIST    List of reserved peers,
+  --secret HEX             Use node key hash
```

Comment:
if this is passed on the CLI, won't it be visible to anyone able to run `ps`? maybe better to pass via stdin/out.